doc/feature/transferFunctions.md

```diff
@@ -0,0 +1,238 @@
+Transfer functions {#transferFunctions}
+============
+
+Being able to specify a mapping between scalar variables and parameters used
+for represent visually those variable is a recurring theme in visualization
+applications. An exemplary case is the transfer functions that map density to
+RGA and opacity values in volume rendering with an emission-absorption optical
+model.
+
+This specification describes the messages and C++ helper classes to deal with
+transfer functions (TFs), considering the usual RGBA as a particular case.
+
+## Requirements
+
+- RGBA colormaps need to be considered first class citizens.
+- However, some applications require more attributes that just RGBA, so it
+  should be possible to describe TFs that map a variable to arbitrary attributes.
+- The C++ helper class will encapsulate a colormap and allow the conversion of
+  the network representation into a OpenGL ready lookup table for creating a
+  texture.
+- The network representation shouldn't restrict the options in the
+  implementation of a GUI editor.
+
+## Design
+
+The consensus has been opting for using 2D piece-wise linear functions specified
+as (unsorted) lists of control points in R^2. An RGBA colormap is an aggregation
+of 4 of this control point lists. Generic TFs are described with a list of
+control points and the name of the target parameter.
+
+Control points are always in the absolute scale, that means that the x coordinate
+refers to the scale of the source variable and the y coordinate refers to the
+scale of the target attribute. The range of RGBA channels will be [0, 1] unless
+an application specifies something different (e.g. in physically based TFs)
+For example, a simple black to white colormap
+for neuron membrane voltage
+could be described by the control points (-80, 0), (-10, 1)
+
+Transfer functions or colormaps cannot be partially updated, the messages
+always describe the entire function. In cases where the graphical editor allows
+remapping a color map to a different input range the editor is responsible
+of finding the adjusted x positions of the control points. For convenience, the
+editor may store colormaps using normalized coordinates for the x axis and then
+recalculate the positions when applied to an application.
+
+### Messages
+
+    namespace lexis.render.detail; // subnamespace needed to avoid name clashes
+
+    table ControlPoint
+    {
+        x: float;
+        y: float;
+    }
+
+    table TransferFunction
+    {
+        attribute: string;
+        points: [ControlPoint];
+    }
+
+    table ColorMap
+    {
+        red: [ControlPoint];
+        green: [ControlPoint];
+        blue: [ControlPoint];
+        alpha: [ControlPoint];
```

Comment:
This apparent redundancy is on purpose.
A ColorMap has very well defined semantics, which if course you can define with based on TransferFunction, but it would be less clear and more error prone. For example, since you cannot enforce any value for the attribute member of the nested TransferFunctions, that would allow inconsistencies in the ColorMap message (repeated channels, wrong channel names). I prefer not writing an issue for this because the requirements motivate the existence of ColorMap as a concept on its own. Many applications would use ColorMap and not use TransferFunction at all.